Core/src/main/java/com/plotsquared/core/util/PlayerManager.java

```diff
@@ -164,25 +164,51 @@ public static void getUUIDsFromString(
      * @return The player's name, None, Everyone or Unknown
      */
     public static @NonNull String getName(final @Nullable UUID owner) {
-        return getName(owner, true);
+        return getName(owner, LocaleHolder.console());
+    }
+
+    /**
+     * Get the name from a UUID.
+     *
+     * @param owner        Owner UUID
+     * @param localeHolder The locale holder to use as the translation provider
+     * @return The player's name, None, Everyone or Unknown
+     */
+    public static @NonNull String getName(final @Nullable UUID owner, final @NonNull LocaleHolder localeHolder) {
+        return getName(owner, localeHolder, true);
     }
 
     /**
      * Get the name from a UUID.
      *
      * @param owner    Owner UUID
-     * @param blocking Whether or not the operation can be blocking
+     * @param blocking Whether the operation can be blocking
      * @return The player's name, None, Everyone or Unknown
      */
     public static @NonNull String getName(final @Nullable UUID owner, final boolean blocking) {
+        return getName(owner, LocaleHolder.console(), blocking);
+    }
+
+    /**
+     * Get the name from a UUID.
+     *
+     * @param owner        Owner UUID
+     * @param localeHolder The locale holder to use as the translation provider
+     * @param blocking     Whether the operation can be blocking
+     * @return The player's name, None, Everyone or Unknown
+     */
+    public static @NonNull String getName(
+            final @Nullable UUID owner, final @NonNull LocaleHolder localeHolder,
+            final boolean blocking
+    ) {
         if (owner == null) {
-            TranslatableCaption.of("info.none");
+            return TranslatableCaption.of("info.none").getComponent(localeHolder);
```

Comment:
Rather than returning the error string, might be better just to send the error message directly to the player, and adjust this method and its callees by returning null or empty? The previous behaviour was returning null after all

Reply:
The previous behaviour basically ignored the `@NonNull` contract and was wrong therefore. Changing a method from `@NonNull` to `@Nullable` changes its contract in an unsafe manner and should be avoided.

Directly sending the message from this method should be avoided, because that method can be used to show the name in separate places: embedded in different messages, sent as action bar, etc.

That said, the method should probably be deprecated, with a new method returning a `Caption`. This way, the `LocaleHolder` is not required here. WDYT?

Reply:
I mean, the contract is already not fullfilled. Until now, `return name` is called in either case, even if name is null.